src/cache/tlru-cache.ts

```diff
@@ -32,32 +38,65 @@ export class TLRUCache extends LRUCache {
   }
 
   private evictExpiredCacheEntries() {
-    const now = new Date(Date.now());
     let cacheKey: string;
-    let evictionDate: Date;
 
     // Not using this.cache.forEach so we can break the loop once
     // we find the fist non-expired entry. Each entry after that
-    // is guaranteed to also be non-expired, because they are oldest->newest
-    for ([cacheKey, evictionDate] of this.cacheEntriesTTLRegistry.entries()) {
-      if (now >= evictionDate) {
+    // is guaranteed to also be non-expired, because iteration happens
+    // in insertion order
+    for (cacheKey of this.cache.keys()) {
+      if (!this.isCacheEntryValid(cacheKey)) {
         this.delete(cacheKey);
       } else {
         break;
       }
     }
   }
 
+  entries(): IterableIterator<[string, string]> {
+    this.evictExpiredCacheEntries();
+    return super.entries();
+  }
+
+  keys(): IterableIterator<string> {
+    this.evictExpiredCacheEntries();
+    return super.keys();
+  }
+
+  values(): IterableIterator<string> {
+    this.evictExpiredCacheEntries();
+    return super.values();
+  }
+
   delete(key: string): boolean {
     this.clearCacheEntryEvictionTimeIfExists(key);
     return super.delete(key);
   }
 
+  // has(key: string): boolean {
```

Comment:
Since `has()` is public, I think you do need to override this to check expiration